### DISCUSSION TITLE
is React.memo necessary if we do not want to render child component?

### DISCUSSION THREAD

**Discussion by Rongjianying** [pmndrs/valtio#602] [Q&A] [answered]
For example, there is a proxy object like:
```javascript
const testStore = proxy({ a: 1 });
```
and there are two components, one parent and one child:
```javascript
const Child = () => {
  return <>child</>;
};

const Parent = () => {
  const testSnap = useSnapshot(testStore);
  return (
    <>
      <Button
          onClick={() => {
            ++testStore.a;
          }}
      >
        {`testStore.a: ${testSnap.a}`}
      </Button>
      <Child/>
    </>
  )
}
```
When I click the button to update `testStore.a`, the `Parent` component is rerendered. But the `Child` component is also rerendered which is not needed.
Is it necessary to use React.memo in this case? Or is there some way in Valtio?

Thx!!!

**Comment by dai-shi** [comment, marked as answer]
Yes, but be careful not to pass snap objects to child component props.

**Reply by Rongjianying** [reply]
thank you very much~~
